towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+            self._logger.error("Error updating the job template's project.")
+        return response
+
+    def change_project_of_job_template_to_prod_branch_project(self, job_template, scm_url):
+        """A job template in the ansible tower has to have a project from the list of projects.
+
+        This function helps to change the project name of a job template in ansible tower for a given job template.
+
+        Args:
+            job_template: given job template object
+            scm_url: the scm url of the projects.
+
+        Returns:
+            Object: The updated job_template object
+
+        """
+        prod_branch_project_id = self.get_production_branch_project_id(scm_url)
+        response = self.change_project_of_job_template(job_template, prod_branch_project_id)
+        return response
+
+    def change_job_type(self, given_labels, new_job_type):
+        """Change the job template type in the given Ansible Tower, which means whether a job should run in 'Run' or
+        'Check' mode.
+
+        Args:
+            given_labels: ist of labels which need to match
+            new_job_type: the updated job type.
+
+        Returns:
+
+        """
+        count = 0
+        job_templates = self.get_all_job_templates()
+        if job_templates is None:
+            self._logger.error("Error finding job templates.")
+        else:
+            for job_template in job_templates:
+                if job_template.labels['count'] > 0:
+                    label_results = job_template.labels['results']
+                    job_labels = [item['name'] for item in label_results]
+                    if set(given_labels) == set(job_labels):
+                        count += 1
+                        self._logger.info("Equal labels found for the job template '{}'".format(job_template.name))
+                        existing_job_type = job_template.job_type
+                        if existing_job_type.lower() == new_job_type.lower():
+                            self._logger.info(
+                                "Existing job-type is already: '{}'. Change is not required for this job template.".
+                                    format(existing_job_type))
+                        else:
+                            self._logger.debug("Changing job type from '{}' to '{}'"
+                                               .format(existing_job_type, new_job_type))
+                            job_template_data = {
+                                "id": job_template.id,
+                                "job_type": new_job_type,
+                                "name": job_template.name,
+                                "playbook": job_template.playbook
+                            }
+                            self.change_job_template_data(job_template_data)
+            if count < 1:
+                self._logger.debug("No job template, which matched all the labels in the config file.")
+
+    def project_exists(self, project_name):
```

Comment:
This is already implemented in the EntityManager with the contains magic method. 
```project_name in self.projects```